spaceship/lib/spaceship/upgrade_2fa_later_client.rb

```diff
@@ -0,0 +1,91 @@
+require_relative 'globals'
+require_relative 'tunes/tunes_client'
+
+module Spaceship
+  class Client
+    def try_upgrade_2fa_later(response)
+      if ENV['SPACESHIP_SKIP_2FA_UPGRADE'].nil?
+        puts("Skipping automatic bypass of 2FA upgrade")
+        return false
+      end
+
+      puts("This account is being prompted to upgrade to 2FA")
+      puts("Attempting to automatically bypass the upgrade until a later date")
+      puts("To disable this, remove SPACESHIP_SKIP_2FA_UPGRADE=1 environment variable")
+
+      # Get URL that requests a repair and gets the widget key
+      widget_key_location = response.headers['location']
+      uri    = URI.parse(widget_key_location)
+      params = CGI.parse(uri.query)
+
+      widget_key = (params['widgetKey'] || []).first
+      if widget_key.nil?
+        STDERR.puts("Couldn't find widgetKey to continue with requests")
+        return false
+      end
+
+      # Step 1 - Request repair
+      response_repair = request(:get) do |req|
+        req.url(widget_key_location)
+      end
+
+      # Step 2 - Request repair options
+      response_repair_options = request(:get) do |req|
+        req.url("https://appleid.apple.com/account/manage/repair/options")
+
+        req.headers['scnt'] = response_repair.headers['scnt']
+        req.headers['X-Apple-Id-Session-Id'] = response.headers['X-Apple-Id-Session-Id']
+        req.headers['X-Apple-Session-Token'] = response.headers['X-Apple-Repair-Session-Token']
+
+        req.headers['X-Apple-Skip-Repair-Attributes'] = '[]'
+        req.headers['X-Apple-Widget-Key'] = widget_key
+
+        req.headers['Content-Type'] = 'application/json'
+        req.headers['X-Requested-With'] = 'XMLHttpRequest'
+        req.headers['Accept'] = 'application/json, text/javascript'
+      end
+
+      # Step 3 - Request setup later
+      request(:get) do |req|
+        req.url("https://appleid.apple.com/account/security/upgrade/setuplater")
+
+        req.headers['scnt'] = response.headers['scnt']
+        req.headers['X-Apple-Id-Session-Id'] = response.headers['X-Apple-Id-Session-Id']
+        req.headers['X-Apple-Session-Token'] = response_repair_options.headers['x-apple-session-token']
```

Comment:
Did you check if it works if you use the `scnt` and `X-Apple-Id-Session-Id` from the `response_repair_options`? I noticed that most of these calls return the same headers, so I would think it would be safest to take the values that you got from the latest response just in case they can change between calls

Reply:
Ah, it does work with that `scnt` 😊  I will change it!